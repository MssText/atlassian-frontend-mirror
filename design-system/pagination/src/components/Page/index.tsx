import React, { Component } from 'react';

import {
  Appearance,
  CustomThemeButton,
  CustomThemeButtonProps,
} from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  CustomThemeButtonProps,
  {
    appearance?: Appearance;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <CustomThemeButton {...this.props} appearance="subtle" />;
  }
}
