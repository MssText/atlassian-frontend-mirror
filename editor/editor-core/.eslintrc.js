module.exports = {
  overrides: [
    {
      // Temporary override to apply tokens lint rules to a subset of editor packages
      // TODO remove in https://product-fabric.atlassian.net/browse/DSP-4001
      files: ['./src/ADD_PATH_HERE/**/*'],
      rules: {
        '@atlaskit/design-system/ensure-design-token-usage': [
          'error',
          { shouldEnforceFallbacks: true },
        ],
        '@atlaskit/design-system/no-unsafe-design-token-usage': [
          'error',
          { shouldEnforceFallbacks: true },
        ],
        '@atlaskit/design-system/no-deprecated-design-token-usage': ['warn'],
      },
    },
  ],
};
