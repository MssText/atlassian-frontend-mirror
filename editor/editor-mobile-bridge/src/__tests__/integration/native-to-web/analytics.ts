import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  callNativeBridge,
  clearBridgeOutput,
  clearEditor,
  editor,
  skipBrowsers as skip,
  getBridgeOutput,
} from '../_utils';

import { INPUT_METHOD, AnalyticsEventPayload } from '@atlaskit/editor-core';

const initPage = async (browser: Page) => {
  await browser.goto(editor.path);
  await browser.waitForSelector(editor.placeholder);
};

const expectTrackEventsToMatchCustomSnapshot = async (
  browser: any,
  testName: string,
) => {
  const outputEvents = await getBridgeOutput(
    browser,
    'analyticsBridge',
    'trackEvent',
  );

  const trackEvents = outputEvents
    .map((outputEvent: any) => JSON.parse(outputEvent.event))
    .filter(
      (analyticsEvent: AnalyticsEventPayload) =>
        analyticsEvent.eventType === 'track',
    );

  expect(trackEvents).toMatchCustomSnapshot(testName);
};

const simpleBrowserTestCase = async (
  browser: Page,
  testName: string,
  fnName: any,
  ...args: any[]
) => {
  await callNativeBridge(browser, fnName, ...args);
  await expectTrackEventsToMatchCustomSnapshot(browser, testName);
  await clearEditor(browser);
  await clearBridgeOutput(browser);
};

BrowserTestCase(
  'editor: ensure actions fire an analytics event via the bridge',
  { skip },
  async (client: any) => {
    const browser = new Page(client);
    await initPage(browser);

    await simpleBrowserTestCase(
      browser,
      'editor: toggling bold style fires an analytics event via the bridge',
      'onBoldClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      browser,
      'editor: toggling italic style fires an analytics event via the bridge',
      'onItalicClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: toggling underline style fires an analytics event via the bridge',
      'onUnderlineClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: toggling code style fires an analytics event via the bridge',
      'onCodeClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: toggling strike style fires an analytics event via the bridge',
      'onStrikeClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: toggling superscript style fires an analytics event via the bridge',
      'onSuperClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: toggling subscript style fires an analytics event via the bridge',
      'onSubClicked',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: updating status fires an analytics event via the bridge',
      'onStatusUpdate',
      'test-text',
      'red',
      'test-uuid',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: setting block type to heading fires an analytics event via the bridge',
      'onBlockSelected',
      'heading1',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting ordered list fires an analytics event via the bridge',
      'onOrderedListSelected',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting bullet list fires an analytics event via the bridge',
      'onBulletListSelected',
      INPUT_METHOD.TOOLBAR,
    );

    await callNativeBridge(
      browser,
      'onBulletListSelected',
      INPUT_METHOD.TOOLBAR,
    );
    await browser.type(editor.placeholder, 'lol');
    await browser.keys(['Enter']);
    await clearBridgeOutput(browser);
    await callNativeBridge(browser, 'onIndentList', INPUT_METHOD.TOOLBAR);
    await expectTrackEventsToMatchCustomSnapshot(
      browser,
      'editor: indenting list fires analytics events via the bridge',
    );
    await clearEditor(browser);
    await clearBridgeOutput(browser);

    await callNativeBridge(
      browser,
      'onBulletListSelected',
      INPUT_METHOD.TOOLBAR,
    );
    await clearBridgeOutput(browser);
    await callNativeBridge(browser, 'onOutdentList', INPUT_METHOD.TOOLBAR);
    await expectTrackEventsToMatchCustomSnapshot(
      browser,
      'editor: outdenting list fires analytics events via the bridge',
    );
    await clearEditor(browser);
    await clearBridgeOutput(browser);

    await simpleBrowserTestCase(
      client,
      'editor: inserting link fires analytics events via the bridge',
      'onLinkUpdate',
      'test-link-title',
      'https://test.link.url',
      INPUT_METHOD.TOOLBAR,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting block quote fires an analytics event via the bridge',
      'insertBlockType',
      'blockquote',
      INPUT_METHOD.INSERT_MENU,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting code block fires an analytics event via the bridge',
      'insertBlockType',
      'codeblock',
      INPUT_METHOD.INSERT_MENU,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting panel fires an analytics event via the bridge',
      'insertBlockType',
      'panel',
      INPUT_METHOD.INSERT_MENU,
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting action fires an analytics event via the bridge',
      'insertBlockType',
      'action',
      INPUT_METHOD.INSERT_MENU,
      'test-action-list-id',
      'test-action-item-id',
    );

    await simpleBrowserTestCase(
      client,
      'editor: inserting decision fires an analytics event via the bridge',
      'insertBlockType',
      'decision',
      INPUT_METHOD.INSERT_MENU,
      'test-decision-list-id',
      'test-decision-item-id',
    );
  },
);
