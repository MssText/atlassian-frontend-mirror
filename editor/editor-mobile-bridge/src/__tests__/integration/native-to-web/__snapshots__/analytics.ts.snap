// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editor: indenting list fires analytics events via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "indentation",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "direction": "indent",
    "indentType": "list",
    "inputMethod": "toolbar",
    "newIndentLevel": 2,
    "nodeLocation": "listItem",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousIndentationLevel": 1,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: indenting list fires analytics events via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "indentation",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "direction": "indent",
    "indentType": "list",
    "inputMethod": "toolbar",
    "newIndentLevel": 2,
    "nodeLocation": "listItem",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousIndentationLevel": 1,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting action fires an analytics event via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "action",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "listLocalId": "test-action-list-id",
    "listSize": 1,
    "localId": "test-action-item-id",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "position": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
    "userContext": "new",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting action fires an analytics event via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "action",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "listLocalId": "test-action-list-id",
    "listSize": 1,
    "localId": "test-action-item-id",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "position": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
    "userContext": "new",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting block quote fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "blockQuote",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting block quote fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "blockQuote",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting bullet list fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "bulletedList",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting bullet list fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "bulletedList",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting code block fires an analytics event via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "codeBlock",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting code block fires an analytics event via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "codeBlock",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting decision fires an analytics event via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "decision",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "listLocalId": "test-decision-list-id",
    "listSize": 1,
    "localId": "test-decision-item-id",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "position": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
    "userContext": "new",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting decision fires an analytics event via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "decision",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "listLocalId": "test-decision-list-id",
    "listSize": 1,
    "localId": "test-decision-item-id",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "position": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
    "userContext": "new",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting link fires analytics events via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "link",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "fromCurrentDomain": false,
    "inputMethod": "toolbar",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "nonPrivacySafeAttributes": Object {
    "linkDomain": "test.link.url",
  },
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting link fires analytics events via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "link",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "fromCurrentDomain": false,
    "inputMethod": "toolbar",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "nonPrivacySafeAttributes": Object {
    "linkDomain": "test.link.url",
  },
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting ordered list fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "numberedList",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting ordered list fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "numberedList",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting panel fires an analytics event via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "panel",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "panelType": "info",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: inserting panel fires an analytics event via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "panel",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "insertMenu",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "panelType": "info",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: outdenting list fires analytics events via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "indentation",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "direction": "outdent",
    "indentType": "list",
    "inputMethod": "toolbar",
    "newIndentLevel": 0,
    "nodeLocation": "listItem",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousIndentationLevel": 1,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: outdenting list fires analytics events via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "indentation",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "direction": "outdent",
    "indentType": "list",
    "inputMethod": "toolbar",
    "newIndentLevel": 0,
    "nodeLocation": "listItem",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousIndentationLevel": 1,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: setting block type to heading fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "heading",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "newHeadingLevel": 1,
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousHeadingLevel": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: setting block type to heading fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "heading",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "newHeadingLevel": 1,
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "previousHeadingLevel": 0,
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling bold style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "strong",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling bold style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "strong",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling code style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "code",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling code style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "code",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling italic style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "italic",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling italic style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "italic",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling strike style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "strike",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling strike style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "strike",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling subscript style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "subscript",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling subscript style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "subscript",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling superscript style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "superscript",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling superscript style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "superscript",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling underline style fires an analytics event via the bridge 1`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "underline",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: toggling underline style fires an analytics event via the bridge 2`] = `
Object {
  "action": "formatted",
  "actionSubject": "text",
  "actionSubjectId": "underline",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "nodeLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: updating status fires an analytics event via the bridge 1`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "status",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;

exports[`editor: updating status fires an analytics event via the bridge 2`] = `
Object {
  "action": "inserted",
  "actionSubject": "document",
  "actionSubjectId": "status",
  "attributes": Object {
    "appearance": "mobile",
    "componentName": "editorCore",
    "inputMethod": "toolbar",
    "insertLocation": "doc",
    "packageName": "@atlaskit/editor-core",
    "packageVersion": "999.9.9",
    "selectionPosition": "start",
    "selectionType": "cursor",
  },
  "eventType": "track",
  "source": "unknown",
  "tags": Array [
    "editor",
    "fabricEditor",
  ],
}
`;
